### ISSUE TITLE
[rush] Only rebuild packages whose dependencies have changed.

### ISSUE THREAD

**Issue by iclanton** [microsoft/rushstack#1546]
**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**

When any dependencies anywhere in the repo are changed, `rush build` rebuilds app projects.

**What is the expected behavior?**

Only rebuild projects that are affected by the changed dependencies.

**Comment by iclanton** [comment]
This issue is addressed here: https://github.com/microsoft/web-build-tools/pull/1526

Right now, the `common/config/<shrinkwrap>` file is included in the `package-deps.json` file (the file used to track files inside projects that have changed). This change drops a `project-dependencies.json` in the `<project>/.rush/temp` folder and instead includes that file in the `package-deps.json` file.

The `project-dependencies.json` is generated during `rush link` by reading and recording the integrity fields of the temp projects' dependencies listed in the shrinkwrap, and recusing and recording the dependencies' dependencies. The file that gets dropped looks like this:

```
{
  "@types/node@8.10.54": "sha512-kaYyLYf6ICn6/isAyD4K1MyWWd5Q3JgH6bnMN089LUx88+s4W8GvK9Q6JMBVu5vsFFp7pMdSxdKmlBXwH/VFRg==",
  "colors@1.2.5": "sha512-erNRLao/Y3Fv54qUa0LBB+//Uf3YwMUmdJinN20yMXm9zdKKqH9wt7R9IIVZ+K7ShzfpLV/Zg8+VyrBJYB4lpg==",
  "gulp@4.0.2": "sha512-dvEs27SCZt2ibF29xYgmnwwCYZxdxhQ/+LFWlbAW8y7jt68L/65402Lz3+CKy0Ov4rOs+NERmDq7YlZaDqUIfA=="
}
```

This becomes a flattened mapping of the package specifier (for example, `gulp@4.0.2`) to the package integrity from the shrinkwrap.

**Comment by iclanton** [comment]
This has been released in Rush 5.15.0
